### PR TITLE
🌱 test for server image list condition

### LINKED ISSUE #943: Test whether conditions are set
## Tasks:
- [x] Test https://github.com/syself/cluster-api-provider-hetzner/pull/942
- [ ] Search for other conditions and see whether they are being tested or not. Make a list and post it as comment to this issue
- [ ] Implement tests

### PR DESCRIPTION
**Pull request by aniruddha2000**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #943 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squash commits
- [ ] include documentation
- [ ] add unit tests

### CODE REVIEW

**Review by janiskemper — COMMENTED** [review]
what happened to these tests that have to manually set the bootstrap condition? Why did you change something here?

File: controllers/hcloudmachine_controller_test.go [hunk 1]
```diff
@@ -161,90 +161,159 @@ var _ = Describe("HCloudMachineReconciler", func() {
...
-				return isPresentAndTrue(key, hcloudMachine, infrav1.BootstrapReadyCondition)
-			}, timeout, time.Second).Should(BeTrue())
+				Eventually(func() bool {
+					return isPresentAndTrue(key, hcloudMachine, infrav1.ServerAvailableCondition)
+				}, timeout, time.Second).Should(BeTrue())
+			})
+		})
 
-			By("listing hcloud servers")
+		Context("correct server", func() {
```
Comment (aniruddha2000) [inline controllers/hcloudmachine_controller_test.go:258]: change the name

File: controllers/hcloudmachine_controller_test.go [hunk 2]
```diff
@@ -133,118 +133,162 @@ var _ = Describe("HCloudMachineReconciler", func() {
...
 
-			By("checking if server available condition is set")
+			AfterEach(func() {
+				Expect(testEnv.Cleanup(ctx, hcloudMachine, hetznerCluster)).To(Succeed())
+			})
 
-			Eventually(func() bool {
-				return isPresentAndTrue(key, hcloudMachine, infrav1.ServerAvailableCondition)
-			}, timeout, time.Second).Should(BeTrue())
+			It("checks for ServerCreateSucceededCondition of hcloudMachine object", func() {
```
Comment (janiskemper) [inline controllers/hcloudmachine_controller_test.go:287]: It checks that ImageNotFound is visible in conditions if image does not exist

File: pkg/services/hcloud/server/server.go [hunk 3]
```diff
@@ -480,6 +480,7 @@ func (s *Service) getServerImage(ctx context.Context) (*hcloud.Image, error) {
 	}
 
 	images = append(images, imagesByName...)
+	s.scope.Logger.Info("lenth of image", "len", len(images), "imagesByName", len(imagesByName), "actual name", s.scope.HCloudMachine.Spec.ImageName)
```
Comment (janiskemper) [inline pkg/services/hcloud/server/server.go:483]: rm

**Review by janiskemper — CHANGES_REQUESTED** [review]

**Review by janiskemper — APPROVED** [review]

**Pull request merged**